client/extensions/woocommerce/state/sites/settings/email/test/selectors.js

```diff
@@ -0,0 +1,214 @@
+							settingsRequest: false,
+							settingsRequestError: false,
+							syncStatus: {
+								last_updated_time: '2017-10-17T22:22:44',
+								store_syncing: false,
+								mailchimp_total_products: 32,
+								product_count: 32,
+								mailchimp_total_orders: 1,
+								order_count: 1,
+								account_name: 'ski',
+								mailchimp_list_name: 'ski',
+								store_id: 'a264ea0b6cbbb',
+							},
+							syncStatusRequest: false,
+							syncStatusRequestError: false,
+							resyncRequest: false,
+							resyncRequestError: false,
+							apiKeySubmit: false,
+							apiKeySubbmitError: false,
+							storeInfoSubmit: false,
+							storeInfoSubmitError: false,
+							listsRequest: false,
+							listsRequestError: false,
+							newsletterSettingsSubmit: false,
+							newsletterSettingsSubmitError: false,
+						},
+					},
+				},
+			},
+		},
+	},
+};
+
+const loadingSettings = Object.assign( {}, emailState, {
+	extensions: {
+		woocommerce: {
+			sites: {
+				123: { settings: { email: { settingsRequest: true } } },
+			},
+		},
+	},
+} );
+
+const submittingApiKey = Object.assign( {}, emailState, {
+	extensions: {
+		woocommerce: {
+			sites: {
+				123: { settings: { email: { apiKeySubmit: true } } },
+			},
+		},
+	},
+} );
+
+const invalidKey = Object.assign( {}, emailState, {
+	extensions: {
+		woocommerce: {
+			sites: {
+				123: { settings: { email: { settings: wrongKeyResponse } } },
+			},
+		},
+	},
+} );
+
+const mailChimpNoSync = Object.assign( {}, emailState, {
+	extensions: {
+		woocommerce: {
+			sites: {
+				123: { settings: { email: { syncStatus: { mailchimp_list_name: 'n/a' } } } },
+			},
+		},
+	},
+} );
+
+describe( 'selectors', () => {
+	describe( '#isRequestingSettings', () => {
+		test( 'should be false when woocommerce state is not available.', () => {
+			expect( isRequestingSettings( {}, 123 ) ).to.be.false;
+		} );
+
+		test( 'should be false when woocommerce state is loaded.', () => {
+			expect( isRequestingSettings( emailState, 123 ) ).to.be.false;
+		} );
+
+		test( 'should be true when settings settings are currently being fetched.', () => {
+			expect( isRequestingSettings( loadingSettings, 123 ) ).to.be.true;
+		} );
+	} );
+
+	describe( '#mailChimpSettings', () => {
+		test( 'should be empty when woocommerce state is not available.', () => {
+			expect( mailChimpSettings( {}, 123 ) ).to.be.deep.equal( {} );
+		} );
+
+		test( 'should return settings if they are storred in state.', () => {
+			expect( mailChimpSettings( emailState, 123 ) ).to.be.deep.equal( settings );
+		} );
+	} );
+
+	describe( '#requestingSettingsError', () => {
+		test( 'should be null when woocommerce state is not available.', () => {
+			expect( requestingSettingsError( {}, 123 ) ).to.be.null;
+		} );
+
+		test( 'should be false if settings are storred in the state.', () => {
+			expect( requestingSettingsError( emailState, 123 ) ).to.be.fales;
+		} );
+	} );
+
+	describe( '#isSubbmittingApiKey', () => {
```

Comment:
The actual selector name should be `isSubmittingApiKey`, so the tests here will need updated in addition to the selectors themselves and the code calling them. Sorry for missing that before.